substrate/frame/revive/src/vm/evm/instructions/contract/call_helpers.rs

```diff
@@ -64,25 +68,48 @@ pub fn resize_memory<'a, E: Ext>(
 	Some(offset..offset + len)
 }
 
-/// Calculates gas cost and limit for call instructions.
 #[inline]
 pub fn calc_call_gas<'a, E: Ext>(
 	interpreter: &mut Interpreter<crate::vm::evm::EVMInterpreter<'a, E>>,
-	account_load: StateLoad<AccountLoad>,
-	has_transfer: bool,
-	local_gas_limit: u64,
+	callee: Address,
+	scheme: CallScheme,
+	input_len: usize,
+	value: U256,
 ) -> Option<u64> {
-	let call_cost =
-		revm_gas::call_cost(interpreter.runtime_flag.spec_id(), has_transfer, account_load);
-	gas_legacy!(interpreter, call_cost, None);
+	let callee: H160 = callee.0 .0.into();
+	let precompile = <AllPrecompiles<E::T>>::get::<E>(&callee.as_fixed_bytes());
 
-	// EIP-150: Gas cost changes for IO-heavy operations
-	let gas_limit = if interpreter.runtime_flag.spec_id().is_enabled_in(TANGERINE) {
-		// Take l64 part of gas_limit
-		min(interpreter.gas.remaining_63_of_64_parts(), local_gas_limit)
-	} else {
-		local_gas_limit
-	};
+	match precompile {
+		Some(precompile) => {
+			// Base cost depending on contract info
+			let base_cost = if precompile.has_contract_info() {
+				RuntimeCosts::PrecompileWithInfoBase
+			} else {
+				RuntimeCosts::PrecompileBase
+			};
+			gas!(interpreter, base_cost, None);
 
-	Some(gas_limit)
+			// Cost for decoding input
+			gas!(interpreter, RuntimeCosts::PrecompileDecode(input_len as u32), None);
```

Comment:
why don't we charge for RuntimeCosts::CopyFromContract(input_len) for the None branch?

Reply:
Added